src/member.cc

```diff
@@ -1010,6 +1049,23 @@ int RedisModule_OnLoad(RedisModuleCtx* ctx,
   }
   LOG(INFO) << "GcsMode: " << module.GcsModeString();
 
+  long long master_mode = 0;
+  if (argc > 1) {
+    CHECK_EQ(REDISMODULE_OK,
+             RedisModule_StringToLongLong(argv[0], &master_mode));
+  }
+  switch (master_mode) {
+    case 0:
+      module.SetMasterMode(RedisChainModule::MasterMode::kRedis);
+      break;
+    case 1:
+      module.SetMasterMode(RedisChainModule::MasterMode::kEtcd);
+      break;
+    default:
+      return REDISMODULE_ERR;
+  }
+  LOG(INFO) << "GcsMode: " << module.GcsModeString();
```

Comment:
Note that parsing + setting this enum has the same code structure as the above.  This is also the case with the few functions related to the enums in the class declaration.  It'd be great to somehow refactor them to reduce code duplication; but if not, no big deal.

Reply:
I think that passing the enum values in directly from the command line (e.g. `--loadmodule libmaster.so 0 --port 6370`) is hard to understand and doesn't provide a lot of value to the user. What does that 0 mean? You'd have to read the code to find out anyways.

I think that what makes more sense is to add real flags with string arguments (this makes it easy to support legacy use of the actual ints) and let SetGcsMode handle converting strings to enum types. For example:

```
map<string, GcsMode> gcs_modes;

auto gcs_mode = ReadString(argv[1]);
module.SetGcsMode(gcs_mode);
```

```
void SetGcsMode(string mode_str) {
  if (mode_str == "normal" || mode_str == "0") {
    gcs_mode_ = GcsMode::kNormal;
  }
  ...
}
```

Let me know if that makes sense to you. Otherwise, I can look into another way to reduce duplication. I think ultimately, coercing raw ints to enums is going to be unavoidably repetitive because enums all have the same value space, but we want to respect that they are technically "disjoint types".